File: src/main/java/org/frc1778/freezylib/logging/MetaField.java
package org.frc1778.freezylib.logging;

import com.google.gson.annotations.Expose;
import org.frc1778.freezylib.util.Measurement;

public class MetaField extends Field {

  @Expose private String value;

  public <T> MetaField(String name, String unit, T value) {
    setName(name);
    setUnit(unit);
    this.value = String.valueOf(value);
    setType(((Object) value).getClass());
  }

  public <T> MetaField(String name, Measurement.Base unit, T value) {
    this(name, unit.getSymbol(), value);
  }

  @Override
  public String getValue() {
    return value;
  }
}

File: src/main/java/org/frc1778/freezylib/logging/MetaField.kt
package org.frc1778.freezylib.logging

import com.google.gson.annotations.Expose
import org.frc1778.freezylib.util.Measurement

class MetaField<T>(name: String, unit: Measurement.Base, value: T) : Field() {

    @Expose
    override lateinit var value: String

    init {
        this.name = name
        this.unit = unit.symbol
        this.value = value.toString()
        setType((value as Any)::class.java)
    }
}
